raiden/waiting.py

```diff
@@ -378,38 +385,76 @@ def wait_for_healthy(
     wait_for_network_state(raiden, node_address, NODE_NETWORK_REACHABLE, retry_timeout)
 
 
-def wait_for_transfer_success(
+class TransferWaitResult(Enum):
+    SECRET_REGISTERED_ONCHAIN = "secret registered onchain"
+    UNLOCKED = "unlocked"
+    UNLOCK_FAILED = "unlock_failed"
+
+
+def wait_for_received_transfer_result(
     raiden: "RaidenService",
     payment_identifier: PaymentID,
     amount: PaymentAmount,
     retry_timeout: float,
-) -> None:  # pragma: no unittest
-    """Wait until a transfer with a specific identifier and amount
-    is seen in the WAL.
+    secrethash: SecretHash,
+) -> TransferWaitResult:  # pragma: no unittest
+    """Wait for the result of a transfer with the specified identifier
+    and/or secrethash. Possible results are onchain secret registration,
+    successful unlock and failed unlock. For a successful unlock, the
+    amount is also checked.
 
     Note:
         This does not time out, use gevent.Timeout.
     """
     log_details = {"payment_identifier": payment_identifier, "amount": amount}
-    found = False
-    while not found:
+    result = None
+    while result is None:
         assert raiden, TRANSPORT_ERROR_MSG
         assert raiden.wal, TRANSPORT_ERROR_MSG
         assert raiden.transport, TRANSPORT_ERROR_MSG
 
         state_events = raiden.wal.storage.get_events()
         for event in state_events:
-            found = (
+            unlocked = (
```

Comment:
You added a tool to wait for a collection of types, why not use it here?

Reply:
It checks only for types (and doesn't wait), if there is a find we would still need to check if the fields match, and then return different results depending on event type. I think the single conditions are shorter and more readable here.